plugins/modules/mount.py

```diff
@@ -817,6 +831,14 @@ def main():
                 module.fail_json(
                     msg="Error making dir %s: %s" % (name, to_native(e)))
 
+            # Set permissions to the newly created mount point.
+            if mode is not None:
+                try:
+                    changed = module.set_mode_if_different(name, mode, changed)
```

Comment:
This does not work on non-Linux operating systems? I've seen you run integration tests only on Linux systems when it was using `chmod`, I don't know about this function.
If so, please also mention this on parameter notes, and maybe warn user when the target os is not Linux?